File: packages/sea-core-ios/Sources/SEACore/SEAEvent.swift
import Foundation

/// A single telemetry event (contract §3.6, spec §20.1).
public struct SEAEvent {
    public let name: String
    public let properties: [String: String]
    public let timestamp: Date

    public init(name: String, properties: [String: String] = [:], timestamp: Date = Date()) {
        self.name = name
        self.properties = properties
        self.timestamp = timestamp
    }
}

/// Host-supplied telemetry sink. `record(_:)` is always called on the main
/// thread.
public protocol SEATelemetrySink: AnyObject {
    func record(_ event: SEAEvent)
}

File: packages/sea-core-android/src/main/java/com/bankerise/sea/core/SEAEvent.kt
package com.bankerise.sea.core

/**
 * A single telemetry event (contract §3.6, spec §20.1).
 */
data class SEAEvent(
    val name: String,
    val properties: Map<String, String> = emptyMap(),
    val timestamp: Long = System.currentTimeMillis()
)

/**
 * Host-supplied telemetry sink. [record] is always called on the main thread.
 */
interface SEATelemetrySink {
    fun record(event: SEAEvent)
}
